src/assets.rs

```diff
@@ -0,0 +1,364 @@
+        Ok(self.store.load(code)?)
+    }
+
+    /// Get a mutable asset editor by the code.
+    pub fn get_mut<L: Ledger>(
+        &mut self,
+        code: &AssetCode,
+    ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        let asset = self.get(code)?;
+        Ok(AssetEditor::new(&mut self.store, asset))
+    }
+
+    /// Commit the store version.
+    pub fn commit<L: Ledger>(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.commit_version()?)
+    }
+
+    /// Revert the store version.
+    pub fn revert<L: Ledger>(&mut self) -> Result<(), KeystoreError<L>> {
+        Ok(self.store.revert_version()?)
+    }
+
+    /// Create an unverified and nontemporary asset.
+    ///
+    /// If the store doesn't have an asset with the same code, adds the created asset to the store.
+    /// Otherwise, updates the exisiting asset.
+    ///
+    /// Returns the editor for the created asset.
+    ///
+    /// To create a verified and temporary asset, use `create_verified` instead.
+    pub fn create<L: Ledger>(
+        &mut self,
+        definition: AssetDefinition,
+        mint_info: Option<MintInfo>,
+    ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        let asset = Asset {
+            definition: definition.clone(),
+            name: None,
+            description: None,
+            icon: None,
+            mint_info,
+            verified: false,
+            temporary: false,
+        };
+        let mut editor = AssetEditor::new(&mut self.store, asset.clone());
+        editor.update();
+        self.store.store(&definition.code, &asset)?;
+        Ok(AssetEditor::new(&mut self.store, asset))
+    }
+
+    /// Create a verified and temporary asset without mint information.
+    ///
+    /// If the store doesn't have an asset with the same code, adds the created asset to the store.
+    /// Otherwise, updates the exisiting asset.
+    ///
+    /// Returns the editor for the created asset.
+    ///
+    /// To create an unverified and nontemporary asset, use `create` instead.
+    fn create_verified<L: Ledger>(
+        &mut self,
+        definition: AssetDefinition,
+    ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        let asset = Asset {
+            definition: definition.clone(),
+            name: None,
+            description: None,
+            icon: None,
+            mint_info: None,
+            verified: true,
+            temporary: true,
+        };
+        let mut editor = AssetEditor::new(&mut self.store, asset.clone());
+        editor.update();
+        self.store.store(&definition.code, &asset)?;
+        Ok(AssetEditor::new(&mut self.store, asset))
+    }
+
+    /// Create a native asset.
+    ///
+    /// Returns the editor for the created asset.
+    fn create_native<L: Ledger>(
+        &mut self,
+        definition: AssetDefinition,
+    ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        let asset = Asset {
```

Comment:
Use `Asset::native()` or `Asset::new_native()` (whatever we decide to call it) instead of duplicating the logic here